terratorch/tasks/classification_tasks.py

```diff
@@ -96,21 +97,36 @@ def __init__(
         """
         self.aux_loss = aux_loss
         self.aux_heads = aux_heads
-        self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+
         super().__init__()
+
+        if model:
+            self.model = model
+
         self.train_loss_handler = LossHandler(self.train_metrics.prefix)
         self.test_loss_handler = LossHandler(self.test_metrics.prefix)
         self.val_loss_handler = LossHandler(self.val_metrics.prefix)
         self.monitor = f"{self.val_metrics.prefix}loss"
 
+        if model_factory and model is None:
+            self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+
     # overwrite early stopping
     def configure_callbacks(self) -> list[Callback]:
         return []
 
     def configure_models(self) -> None:
+
+        if not hasattr(self, "model_factory"):
+            # Custom model is provided
+            if self.hparams["freeze_backbone"] or self.hparams["freeze_decoder"]:
+                logger.warning("freeze_backbone and freeze_decoder are ignored if a custom model is provided.")
+            return
```

Comment:
@Joao-L-S-Almeida doesn't it mean configure_models will always return if model_factory is not set?

Reply:
@romeokienzler That's how it should work. If model_factory is not provided, a custom model is passed. Therefore, the rest of the function should be skipped.